src/data/device_automation.ts

```diff
@@ -9,6 +9,10 @@ export interface DeviceAutomation {
   event?: string;
 }
 
+export interface DeviceAction extends DeviceAutomation {
+  device: string;
```

Comment:
So how does this work ? Why not reuse device_id ? 

Also wondering if we should have these interfaces extend from one another? They represent pretty different things (trigger/condition/action)

Reply:
The reason for extending is so we can assume things about `T`, e.g. that it has a string `device_id` member:
https://github.com/home-assistant/home-assistant-polymer/blob/c37fa7bb5fb0ae9304a442cb9a74d2e0baf465a4/src/components/device/ha-device-automation-picker.ts#L24-L26